### PR TITLE
Osiris ik update

### PR DESCRIPTION
**Pull request by gsn9**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/osirisrex/objs/OsirisRexOcamsCamera/OsirisRexOcamsCamera.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ namespace Isis {
       m_instrumentNameLong = "Sampling Camera";
       m_instrumentNameShort = "SamCam";
     } // IK values for polycam: -64360 (general) and -64616 to -64500 (focus specific)
-    else if (frameCode == -64360) {
+    else if ((frameCode == -64360) || (frameCode >= -64616 && frameCode <= -64500)) {
```
Comment (Kelvinrr) [inline isis/src/osirisrex/objs/OsirisRexOcamsCamera/OsirisRexOcamsCamera.cpp:53]: That's a large range of numbers, what is the significance of the IK codes. 

I'm sure it's correct just curious.

File: isis/tests/OsirisRexOCamsCameraTests.cpp [hunk 2]
```diff
@@ -95,3 +95,41 @@ TEST_F(OsirisRexCube, SamplingCam) {
   EXPECT_NEAR(cam->Line(), 5, 0.01);
   EXPECT_NEAR(cam->Sample(), 5, 0.01);
 }
+
+TEST_F(OsirisRexCube, PolyCamUpdatedIkCodes){
+setInstrument("-64500", "PolyCam");
```
Comment (Kelvinrr) [inline isis/tests/OsirisRexOCamsCameraTests.cpp:100]: weird white space?

File: isis/tests/OsirisRexOCamsCameraTests.cpp [hunk 3]
```diff
@@ -95,3 +95,41 @@ TEST_F(OsirisRexCube, SamplingCam) {
...
+
+  OsirisRexOcamsCamera *cam = (OsirisRexOcamsCamera *)testCube->camera();
+
+  EXPECT_EQ(cam->instrumentRotation()->Frame(), -27002);
+
+  // Test kernel IDs
+  EXPECT_EQ(cam->CkFrameId(), -64000);
+  EXPECT_EQ(cam->CkReferenceId(), 1);
+  EXPECT_EQ(cam->SpkTargetId(), -64);
+  EXPECT_EQ(cam->SpkReferenceId(), 1);
```
Comment (Kelvinrr) [inline isis/tests/OsirisRexOCamsCameraTests.cpp:110]: Probably worth checking the NaifIkCode in the Kernels group

**Comment by jessemapel** [comment]
Make sure you fill out the PR template. In particular, you need to link the issue so that it's connected to this PR.

File: isis/src/osirisrex/objs/OsirisRexOcamsCamera/OsirisRexOcamsCamera.cpp (hunk 1 above)
Comment (jessemapel) [reply isis/src/osirisrex/objs/OsirisRexOcamsCamera/OsirisRexOcamsCamera.cpp:53]: The IK codes are different focus positions for the sensor.

File: isis/tests/OsirisRexOCamsCameraTests.cpp (hunk 3 above)
Comment (jessemapel) [reply isis/tests/OsirisRexOCamsCameraTests.cpp:110]: The camera doesn't set that value. That value is set by the ingestion program

**Review by jessemapel — COMMENTED** [review]
Camera model changes look good, but you also need to modify the ingestion program, [ocams2isis](https://github.com/USGS-Astrogeology/ISIS3/tree/fbc201bb0f0efeab5c3a0f4b95259268654249cc/isis/src/osirisrex/apps/ocams2isis), so that it adds the appropriate IK code to the label. See the original issue for to get that number.

**Comment by KrisBecker** [comment]
Be aware that the original PolyCam code (-64360) needs to remain in the Kernels group NaifFrameCode keyword so that pointing can be determined. The [frames kernel](https://naif.jpl.nasa.gov/pub/naif/pds/pds4/orex/orex_spice/spice_kernels/fk/orx_v14.tf) does not contain a frame definition for each focus position.

In my opinion, ocams2isis should not attempt to deal with focus position. That knowledge should be isolated in the camera model. Also note that the NaifFrameCode is currently being set in the [Kernels translation](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/osirisrex/apps/ocams2isis/OsirisRexOcamsKernels_fit.trn) file so no change should be needed in ocams2isis.

**Comment by gsn9** [comment]
I think I'm a little mixed up on where the updated IK code's need to be added. Nothing in the ocams2isis need's to be changed currently but I should add the appropriate IK codes to the [Kernels translation](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/osirisrex/apps/ocams2isis/OsirisRexOcamsKernels_fit.trn) file?

**Comment by KrisBecker** [comment]
@gsn9 the OSIRIS-REx PolyCam cube labels already contain the focus position. Its the PolyCamFocusPositionNaifId in the Instrument group. So the only change that should be needed here it to handle camera model settings for each PolyCam focus position.

**Comment by jessemapel** [comment]
Fixes #3079

**Comment by jlaura** [comment]
@jessemapel @gsn9 What is the status on this?

**Comment by jessemapel** [comment]
This is on me right now. I'll check it today so it can be closed off.

**Review by jessemapel — APPROVED** [review]
I've compared this against the O-REx version and this matches up with it.

**Comment by jessemapel** [comment]
For future reference, the handling of different polycam positions was already implemented  by setting the ikcode based on the PolyCamFocusPositionNaifId keyword in the instrument group.

**Pull request merged**